File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.rb
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(s, nums) {
    let result = Number.MAX_SAFE_INTEGER
    let sum = 0, left = 0
    for(let i = 0; i < nums.length; i++){
        sum += nums[i]
        while(sum >= s){
            result = Math.min(result, i - left + 1)
            sum -= nums[left++]
        }
    }
    return result === Number.MAX_SAFE_INTEGER ? 0 : result 
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.php
class Solution
{

    /**
     * @param Integer   $target
     * @param Integer[] $nums
     *
     * @return Integer
     */
    function minSubArrayLen($target, $nums)
    {
        $len = PHP_INT_MAX;

        $count = count($nums);
        for ($i = $j = $s = 0; $i < $count;) {
            $s += $nums[$i++];
            while ($s >= $target) {
                $len = min($len, $i - $j);
                $s -= $nums[$j++];
            }
        }

        return $len < PHP_INT_MAX ? $len : 0;
    }
}